app/controllers/user_confirmations_controller.rb

```diff
@@ -0,0 +1,34 @@
+class UserConfirmationsController < DeviseController
+  include Spree::Core::ControllerHelpers::Auth # Needed for access to current_ability, so we can authorize! actions
+
+  # GET /resource/confirmation/new
+  def new
+    build_resource({})
+  end
+
+  # POST /resource/confirmation
+  def create
+    self.resource = resource_class.send_confirmation_instructions(resource_params)
+
+    if is_navigational_format? && successfully_sent?(resource)
+      set_flash_message(:success, :confirmation_sent)
+    elsif is_navigational_format?
+      set_flash_message(:error, :confirmation_not_sent)
+    end
+
+    respond_with_navigational(resource){ redirect_to login_path }
+  end
+
+  # GET /resource/confirmation?confirmation_token=abcdef
+  def show
+    self.resource = resource_class.confirm_by_token(params[:confirmation_token])
+
+    if is_navigational_format? && resource.errors.empty?
+      set_flash_message(:success, :confirmed)
+    elsif is_navigational_format?
```

Comment:
what about moving the `is_navigational_format?` to its own `if` to avoid the duplication? I wonder though what happens when it is not `navigational` though. I haven't gone through the whole PR again and I don't remember what it is, sorry.

Reply:
I didn't write this bit. To be honest, I'm not even sure what `is_navigational_format?` is, I think it's a devise method for rendering different responses.